### PR TITLE
NDRS-1270: Provide historical auction information

### PR DESCRIPTION
**Pull request by darthsiroftardis**
CHANGELOG:

- Altered the `state_get_auction_info` RPC to take a `block-identifier` arg which retrieves historical auction information relevant to that block
- Allowed for default behavior to point to the highest block in the case no `identifier` is provided

### CODE REVIEW

**Review by Fraser999 — COMMENTED** [review]
There is an error message `get-auction-info failed to get last added block` in state.rs which should be updated to only return that if the optional block identifier is `None`.  In the case of `Some`, we should return a different message stating the requested block is not known.

File: client/src/get_auction_info.rs [hunk 1]
```diff
@@ -12,12 +12,13 @@ enum DisplayOrder {
...
     NodeAddress,
     RpcId,
+    BlockIdentifier,
 }
 
 impl<'a, 'b> ClientCommand<'a, 'b> for GetAuctionInfo {
     const NAME: &'static str = "get-auction-info";
     const ABOUT: &'static str =
-        "Retrieves the bids and validators as of the most recently added block";
+        "Retrieves the bids and validators as of a block added to the linear chain";
```
Comment (EdHastingsCasperLabs) [inline client/src/get_auction_info.rs:21]: "Retrieves the bids and validators as of either a specific block (by height or hash), or the most recently added block";

File: node/src/components/rpc_server/rpcs/docs.rs [hunk 2]
```diff
@@ -83,7 +83,7 @@ pub(crate) static OPEN_RPC_SCHEMA: Lazy<OpenRpcSchema> = Lazy::new(|| {
     schema.push_with_optional_params::<GetEraInfoBySwitchBlock>(
         "returns an EraInfo from the network",
     );
-    schema.push_without_params::<GetAuctionInfo>(
+    schema.push_with_optional_params::<GetAuctionInfo>(
         "returns the bids and validators as of the most recently added Block",
```
[the remark this answers is not in the document]
Comment (EdHastingsCasperLabs) [reply node/src/components/rpc_server/rpcs/docs.rs:87]: ABOUT.to_lowercase()

**Review by EdHastingsCasperLabs — APPROVED** [review]
Assuming requested changes to help text is applied.

File: node/src/components/rpc_server/rpcs/docs.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply node/src/components/rpc_server/rpcs/docs.rs]: Updated

**Comment by darthsiroftardis** [comment]
bors merge

**Pull request merged**